File: src/Applications/UMD_Etc/odas_pre_proc/ARGO/REJECTS/read_argo_greylist.m
% Read Argo Grey List text file
% Write wmo, date, var in a text file for fortran
% ftp://ftp.ifremer.fr/ifremer/argo/ar_greylist.txt

clear all

% WMO, VAR, DATE,empty,QC,comment, dac
fname = 'argo_greylist.txt';
fid   = fopen(fname,'r');

tm = 1082;
wmo   = zeros(tm,1);
ivar  = zeros(tm,1);
idate1 = zeros(tm,1);
idate2 = zeros(tm,1);

  tmp = 'a';
  i = 1;
  while i>0,
    line = fgets(fid);
    str  = find(line==',');
    if isempty(str)==1, break; end 
    wmo(i) = str2num(strtrim(line(1:str(1)-1)));  
    tmp = line(str(1)+1:str(2)-1);ivar(i) = 1;   
      if strcmp(tmp,'TEMP')==1, 
        ivar(i) = 1;
      elseif strcmp(tmp,'PSAL')==1, 
        ivar(i) = 2;
      elseif strcmp(tmp,'PRES')==1, 
        ivar(i) = 3;
      end
    idate1(i) = str2num(line(str(2)+1:str(3)-1));    
    tmp = str2num(line(str(3)+1:str(4)-1)); 
    if isempty(tmp)==0,
      idate2(i) = tmp;    
    else
      idate2(i) = 20201231;  
    end
    i=i+1;  
  end
  nrows = i-1;
status = fclose(fid);

% Sort by date
  sv = sortrows([wmo ivar idate1 idate2],1);

% Write new file
  fid = fopen('ARGO_GREYLIST.txt','w');
  for i=1:nrows,
     fprintf(fid,'%-7d%-1d%-8d%-8d\n',[sv(i,1), sv(i,2), sv(i,3), sv(i,4)]);
  end

uwmo = unique(wmo);
save ARGO_GREYLIST.mat uwmo

File: src/Applications/UMD_Etc/odas_pre_proc/ARGO/REJECTS/read_argo_greylist.py
#! /usr/bin/env python


# Read Argo Grey List text file
# Write wmo, date, var in a text file for fortran
# ftp://ftp.ifremer.fr/ifremer/argo/ar_greylist.txt

import numpy as np
import array
import sys
import scipy.io
from datetime import date

#sys.path.append('/gpfsm/dhome/gmaofcst/ODAS/OBS/V2/UTILS/')
sys.path.append('/discover/nobackup/lren1/pre_proc/NRT/UTILS/')

########################################################################

if len(sys.argv) > 3:
  yyyy = sys.argv[1]
  mm  = sys.argv[2]
  dd  = sys.argv[3] 

mm=str(mm).zfill(2)
dd=str(dd).zfill(2)
yyyy=str(yyyy).zfill(4)


today=date.today()
today=today.strftime("%Y%m%d")


# WMO, VAR, DATE,empty,QC,comment, dac
#fname = 'ar_greylist_' + today+'.txt'
fname ='argo_greylist.txt'
# get number of lines in the file
tm = sum(1 for line in open(fname))
print tm
f   = open(fname,'r')
wmo   = np.zeros(tm,int)
ivar  = np.zeros(tm,int)
idate1 = np.zeros(tm,int)
idate2 = np.zeros(tm,int)

  
i = 0
while i>=0:
  line = f.readline()
  ind = [j for j in range(len(line)) if line.startswith(",", j)]
  if len(ind)==0:
    print 'end of file'
    break
  
  wmo[i] = int(line[0:ind[0]])
  tmp=line[ind[0]+1:ind[1]]
  ivar[i]=1
  if tmp =='TEMP':
    ivar[i] = 1

  if tmp=='PSAL':
    ivar[i] = 2

  if tmp=='PRES':
     ivar[i] = 3
 
  idate1[i] = int(line[ind[1]+1:ind[2]])
  tmp = line[ind[2]+1:ind[3]]
  if len(tmp)>1:
     idate2[i] = int(tmp)
  else:
     idate2[i] = int(today)
#  print wmo[i],ivar[i],idate1[i],idate2[i]
  i = i +1
nrows = i
f.close()
# Sort by date

sv = np.column_stack((wmo,ivar,idate1, idate2))
sv = sv[sv[:,0].argsort()]
# Write new file
fout  = open('ARGO_GREYLIST_test.txt','w')
for i in range (0,nrows):
   print sv[j,0],sv[i,1],sv[i,2],sv[i,3]
   fout.write('{:<7d}{:<1d}{:<8d}{:<8d}\n'.format(sv[i,0], sv[i,1], sv[i,2], sv[i,3]))
fout.close

uwmo = np.unique(wmo);
scipy.io.savemat('ARGO_GREYLIST_test.mat', mdict={'uwmo': uwmo})

